recipes/ruby.rb

```diff
@@ -77,8 +77,16 @@
   end
 
   begin
-    chef_gem 'pg' do
-      compile_time true if respond_to?(:compile_time)
+    # allow optional attribute to install specific version of pg gem
+    if defined?(node['postgresql']['pg_gem']['version']).nil?
+      chef_gem 'pg' do
+        compile_time true if respond_to?(:compile_time)
```

Comment:
Instead of having an if statement with 2 different resources you can just add version node['postgresql']['pg_gem']['version'] if node['postgresql']['pg_gem']['version']